formats/protobuf/commonMain/src/kotlinx/serialization/protobuf/internal/ProtobufReader.kt

```diff
@@ -65,14 +65,12 @@ internal class ProtobufReader(private val input: ByteArrayInput) {
         return input.readExactNBytes(length)
     }
 
-    private fun Input.readExactNBytes(bytes: Int): ByteArray {
-        val array = ByteArray(bytes)
-        var read = 0
-        while (read < bytes) {
-            val i = this.read(array, read, bytes - read)
-            if (i == -1) error("Unexpected EOF")
-            read += i
+    private fun Input.readExactNBytes(bytesCount: Int): ByteArray {
+        if (bytesCount > availableBytes) {
+            error("Unexpected EOF, available $availableBytes bytes, requested: $bytesCount")
         }
+        val array = ByteArray(bytesCount)
+        read(array, 0, bytesCount)
```

Comment:
Does `read` always read exactly `bytesCount` bytes?

Reply:
Yes, that's the virtue of having your own BAOS